plugins/lsp/conform-nvim.nix

```diff
@@ -1,33 +1,83 @@
 {
   lib,
-  helpers,
   config,
   pkgs,
   ...
 }:
-with lib;
 let
-  cfg = config.plugins.conform-nvim;
+  inherit (lib) types;
+  inherit (lib.nixvim) defaultNullOpts;
 in
-{
-  options.plugins.conform-nvim = helpers.neovim-plugin.extraOptionsOptions // {
-    enable = mkEnableOption "conform-nvim";
+lib.nixvim.neovim-plugin.mkNeovimPlugin config {
+  name = "conform-nvim";
+  originalName = "conform.nvim";
+  defaultPackage = pkgs.vimPlugins.conform-nvim;
+
+  maintainers = [ lib.maintainers.khaneliman ];
+
+  # TODO: added 2024-08-23 remove after 24.11
+  deprecateExtraOptions = true;
+  optionsRenamedToSettings = [
+    [
+      "formatAfterSave"
+      "lspFallback"
+    ]
+    [
+      "formatOnSave"
+      "lspFallback"
+    ]
+    [
+      "formatOnSave"
+      "timeoutMs"
+    ]
```

Comment:
Re https://github.com/nix-community/nixvim/pull/2070#discussion_r1729802391:

I see you're defining rename modules for these; I think these will fail non-gracefully if users have a string definition for `formatAfterSave` or `formatOnSave` in their configs though, since we're only creating deprecated options for the old submodule options.

I don't think it's (trivially) possible to gracefully handle _both_ the submodule usage and the string usage.

We could either mark the `formatAfterSave` and `formatOnSave` options as removed, so users with either type definition will get an error message. Or we could try and come up with a bespoke implementation of a rename module that covers both types.

Reply:
If we did go down the bespoke rename module route, we could probably have a `types.anything` option\* that _if defined_ checks whether its definition is a string or an attrs, then aliases the resulting config making use of `toSnakeCase`.

_\* Or just keep the existing type I guess..._

If you think this is worth doing I can try and help with the implementation.

Reply:
Sorry, this explains better what you were saying above. I think I had a TODO to come back to this but must have removed it. I started off with a rename for just the parent options and then realized it wouldn't do the rename for the suboptions. Doing both in the same block failed with option doesn't exist errors because the rename happens for the parent option first. 

I'm using strings in my config for these options so I can play around with it a bit. Unless you have an idea in mind for implementation.

Reply:
We could implement _most_ of a rename alias by

1. Keeping the old option declaration (in `extraOptions`)
2. Removing its default 
3. Adding an `optionalAttrs` config, conditional on `options.*.isDefined`
4. Checking for string vs attrs and aliasing as appropriate
5. Also manually implement warnings, preferably complete with `showDefs`

But again, is this option important enough to do this instead of a removal assertion?